### PR TITLE
Add proper error explanation for issue E0703 in Rust compiler.

### PR DESCRIPTION
**Pull request by Polkaverse**
Add long explanation for the E0703 error code
Part of #61137

r? @GuillaumeGomez

### CODE REVIEW

File: src/librustc_error_codes/error_codes/E0703.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+Invalid ABI(Application Binary Interface) used in the code.
```
Comment (GuillaumeGomez) [inline src/librustc_error_codes/error_codes/E0703.md]: Please add a whitespace before the paren.

File: src/librustc_error_codes/error_codes/E0703.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+Erroneous code example:
+
+```compile_fail,E0703
+extern "invalid" fn foo() {} //~ ERROR
+
+fn main() { }
+```
+At present there the few predefined ABI's (like Rust, C, system, etc.)
```
Comment (GuillaumeGomez) [inline src/librustc_error_codes/error_codes/E0703.md]: Please add an empty line before this one.

File: src/librustc_error_codes/error_codes/E0703.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
+Invalid ABI(Application Binary Interface) used in the code.
+
+Erroneous code example:
+
+```compile_fail,E0703
+extern "invalid" fn foo() {} //~ ERROR
```
Comment (GuillaumeGomez) [inline src/librustc_error_codes/error_codes/E0703.md]: No need for the \~ (for info, it's used in the UI tests, no need here). You can just write instead:

`extern "invalid" fn foo() {} // error!`

File: src/librustc_error_codes/error_codes/E0703.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+fn main() { }
+```
+At present there the few predefined ABI's (like Rust, C, system, etc.)
+which we can use in our Rust code. Please verify the ABI from the
+given ABI. For example you can replace the given ABI from 'Rust'.
+
+```
+extern "Rust" fn foo() {} //~ OK!
```
Comment (GuillaumeGomez) [inline src/librustc_error_codes/error_codes/E0703.md]: Same, no need for \~:

`extern "Rust" fn foo() {} // Ok!`

File: src/librustc_error_codes/error_codes/E0703.md [hunk 5]
```diff
@@ -0,0 +1,18 @@
+Invalid ABI(Application Binary Interface) used in the code.
+
+Erroneous code example:
+
+```compile_fail,E0703
+extern "invalid" fn foo() {} //~ ERROR
+
+fn main() { }
```
Comment (GuillaumeGomez) [inline src/librustc_error_codes/error_codes/E0703.md]: Btw, no need to display it in the docs, you can hide the line by starting it with "# ". So please rewrite the code example like this:

```
extern "invalid" fn foo() {} // error!
# fn main() {}
```

File: src/librustc_error_codes/error_codes/E0703.md [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+```
+At present there the few predefined ABI's (like Rust, C, system, etc.)
+which we can use in our Rust code. Please verify the ABI from the
+given ABI. For example you can replace the given ABI from 'Rust'.
+
+```
+extern "Rust" fn foo() {} //~ OK!
+
+fn main() { }
```
Comment (GuillaumeGomez) [inline src/librustc_error_codes/error_codes/E0703.md]: Same as above, let's hide it:

```
extern "Rust" fn foo() {} // Ok!
# fn main() {}
```

**Review by Dylan-DPC-zz — DISMISSED** [review]

File: src/librustc_error_codes/error_codes/E0703.md [hunk 7]
```diff
@@ -0,0 +1,19 @@
+Invalid ABI (Application Binary Interface) used in the code.
+
+Erroneous code example:
+
+```compile_fail,E0703
+extern "invalid" fn foo() {} // error!
+
```
Comment (GuillaumeGomez) [inline src/librustc_error_codes/error_codes/E0703.md]: Please remove this line.

File: src/librustc_error_codes/error_codes/E0703.md [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+# fn main() {}
+```
+
+At present few predefined ABI's (like Rust, C, system, etc.) can be
+used in Rust. Verify that the ABI is predefined. For example you can
+replace the given ABI from 'Rust'.
+
+```
+extern "Rust" fn foo() {} // ok!
+
```
Comment (GuillaumeGomez) [inline src/librustc_error_codes/error_codes/E0703.md]: Please remove this line.

File: src/librustc_error_codes/error_codes/E0703.md [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+Erroneous code example:
+
+```compile_fail,E0703
+extern "invalid" fn foo() {} // error!
+
+# fn main() {}
+```
+
+At present there the few predefined ABI's (like Rust, C, system, etc.)
+which we can use in our Rust code. Please verify the ABI from the
```
[the remark this answers is not in the document]
Comment (PankajChaudhary5) [reply src/librustc_error_codes/error_codes/E0703.md]: Done. Please review

**Pull request merged**